packages/flame/lib/src/collisions/quadtree/quadtree_collision_detection.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flutter/widgets.dart';
+
+/// Collision detection modification to support a Quad Tree broadphase.
+///
+/// Do not use standard [items] list for components. Instead adds all components
+/// into [QuadTreeBroadphase] class.
+class QuadTreeCollisionDetection extends StandardCollisionDetection {
+  QuadTreeCollisionDetection({
+    required Rect mapDimensions,
+    required ExternalBroadphaseCheck broadphaseCheck,
+    required ExternalMinDistanceCheck minimumDistanceCheck,
+    int maxObjects = 25,
+    int maxLevels = 10,
+  }) : super(
+          broadphase: QuadTreeBroadphase<ShapeHitbox>(
+            mainBoxSize: mapDimensions,
+            maxObjects: maxObjects,
+            maxLevels: maxLevels,
+            broadphaseCheck: broadphaseCheck,
+            minimumDistanceCheck: minimumDistanceCheck,
+          ),
+        );
+
+  QuadTreeBroadphase get quadBroadphase => broadphase as QuadTreeBroadphase;
+
+  final _listenerTransform = <ShapeHitbox, VoidCallback>{};
+  final _listenerCollisionType = <ShapeHitbox, VoidCallback>{};
+  final _scheduledUpdate = <ShapeHitbox>{};
+
+  @override
+  void add(ShapeHitbox item) {
+    super.add(item);
+    // ignore: prefer_function_declarations_over_variables
+    final listenerTransform = () {
+      if (item.isMounted) {
+        _scheduledUpdate.add(item);
+      }
+    };
+    final parent = item.parent;
+    if (parent != null && parent is PositionComponent) {
+      parent.position.addListener(listenerTransform);
```

Comment:
This won't work for `ComposedHitboxes`.

Reply:
And also won't work if the parent has rotated and/or scaled if the anchor is not in the center.
Better just add a listenter to the full `transform`.

Reply:
It seems like it won't work for every component-hitbox hierarchy deeper than 2. For example when adding component inside component inside component inside...  (two days later) -  and then adding a hitbox. 
I think we should keep "default" behavior for `add` function, supporting simple hierarchy, but to create additional function like `void addQuadtreeHitbox(ShapeHitbox hitbox, Component transformedComponent)` where the `transformedComponent` is top-level component on which all transformation should be performed.

Reply:
It looks like I found better solution - to subscribe directly to aabb invalidation.